Automattic/newspack-plugin
# fix: add credit meta attribute dependency to. image block effect hook

### All Submissions:






### Changes proposed in this Pull Request:



Closes https://app.asana.com/0/1207720049982626/1207708215353147/f

This PR fixes a problem where adding our credits image block meta would cause an infinite render loop whenever a synced pattern was created.

We fix this by adding a check for updated meta attribute values to the effect hook as well as adding the attribute meta values to the hook dependencies.

![Screenshot 2024-07-08 at 15 25 02](https://github.com/Automattic/newspack-plugin/assets/17905991/1e4acb4d-f9d7-4a41-8784-44240774925a)

### How to test the changes in this Pull Request:

1. Start on a site running WP 6.6 RC, and using Chrome.
2. Add an image block to the editor.
3. Create a synced pattern from the image block (From block options, select `create pattern` and make sure the synced toggle is on as pictured above)
4. On `release` Chrome will freeze up as the block is re-rendered infinitely. On this branch, there should be no problems.

### Other information:
